### PR TITLE
Scroll to Bottom on entering chat

### LINKED ISSUE #125: Starting a new conversation needs to scroll to most recent
Current behaviour is to start at the oldest messages. This is undesirable.

Need to make it jump to the newest. I'm conscious that we will not always to want to jump to the newest as a user may be checkout out history when a user adds a new message.

### LINKED ISSUE #129: Shouldn't need a timer to scroll to most recent
Solution to #125  isn't ideal as it timer and is clumsy. 

Tried to use https://www.didierboelens.com/2019/04/addpostframecallback/ though had error simular to https://github.com/flutter/flutter/issues/33648

### PR DESCRIPTION
**Pull request by gaslitbytech**
Timer is not ideal. Though trying the approaches

https://www.didierboelens.com/2019/04/addpostframecallback/ errored with odd issue simular to https://github.com/flutter/flutter/issues/33648

EDIT: most recent change fixes #129

### CODE REVIEW

**Comment by nickmeinhold** [comment]
@tourdownunder 

I hope you don't mind me butting in - the bug we found here was interesting, I think it was a combination of: 
- there is still no API call in ListView to scroll to an offset (see [#12319 · flutter/flutter](https://github.com/flutter/flutter/issues/12319)) so we are forced to wait till the contents have been laid out and then use `controller.jumpTo(controller.position.maxScrollExtent);`

- We used a timer to do the waiting then agreed a timer was bad and so tried to user `WidgetsBinding.instance.addPostFrameCallback` which would have been fine, except that we came across another kind of weirdness which is that the first time a `StreamBuilder` builds it builds with no data (ie. no messages and so nothing is laid out, see [#16465 · flutter/flutter](https://github.com/flutter/flutter/issues/16465)). 

Adding a post frame callback to call `jumpTo` is done inside the StreamBuilder's build method, after the check for no data, but we agreed we wanted it done once on first build and not every rebuild so I subclassed StatefulWidget and did it in the initState.  What do you think?  

That was a good learning experience, thanks!

File: ios/Flutter/.last_build_id [hunk 1]
```diff
@@ -0,0 +1 @@
+3a1339969e2fa2c75b036702fdbf881a
```
Comment (nickmeinhold) [inline ios/Flutter/.last_build_id]: We should add this to the gitignore and remove from the PR

File: ios/Podfile.lock [hunk 2]
```diff
@@ -423,16 +423,16 @@ SPEC CHECKSUMS:
   abseil: 18063d773f5366ff8736a050fe035a28f635fd27
   AppAuth: 73574f3013a1e65b9601a3ddc8b3158cce68c09d
   BoringSSL-GRPC: db8764df3204ccea016e1c8dd15d9a9ad63ff318
-  cloud_firestore: 11c87f1d0b4193831045a320aed50723b67f2ea3
```
Comment (nickmeinhold) [inline ios/Podfile.lock]: Let's try and get to the bottom of this so we don't keep fighting over the checksums.  There's an explanation of why it happens and a potential solution here: 

https://artsy.github.io/blog/2016/05/03/podspec-checksums/

If you run `pod update` from `ios/` do you get the same checksums as currently in `dev`?

File: lib/chat_page.dart [hunk 3]
```diff
@@ -67,6 +63,43 @@ class _ChatPageState extends State<ChatPage> {
   }
 }
 
+class MessagesList extends StatefulWidget {
```
Comment (nickmeinhold) [inline lib/chat_page.dart:66]: We should add a widget test for this new widget.

**Review by nickmeinhold — COMMENTED** [review]
I think we should add a few small improvements to the PR (added comments for each one) and I'm happy to do them if no-one else wants to, and I'm also happy to approve the PR so we can get it into the `dev` branch if we add issues for the improvements. @tourdownunder if you'd prefer that let me know and I'll approve the PR.

File: ios/Flutter/.last_build_id (hunk 1 above)
Comment (nickmeinhold) [reply ios/Flutter/.last_build_id]: Sorry, I think this was me, I'll put up a fix.

File: ios/Podfile.lock (hunk 2 above)
Comment (nickmeinhold) [reply ios/Podfile.lock]: I guess this was me too - I'll stop trying to blame other people 😄

**Review by nickmeinhold — APPROVED** [review]
I've fixed the problems I caused and added a test so, imo, this PR is good for merging!

**Comment by gaslitbytech** [comment]
Thanks @nickmeinhold I see that the separating the model from the state helped as you suspected.

**Pull request merged**